### DISCUSSION TITLE
How do I use AsyncIOScheduler to execute time-consuming I/O operations without blocking other tasks

### DISCUSSION THREAD

**Discussion by pwallk** [agronholm/apscheduler#999] [Q&A]
### Things to check first

- [X] I have checked that my issue does not already have a solution in the [FAQ](https://apscheduler.readthedocs.io/en/master/faq.html)

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release

### Version

3.10.4

### What happened?

I'm not familiar with asyncio, When I use AsyncIOScheduler in FastAPI to perform time-consuming IO operations, I block other asyncio tasks as well as the fastapi interface

### How can we reproduce the bug?

```python
async def init_scheduler(app: FastAPI):
    scheduler = AsyncIOScheduler(
        job_defaults={
            'coalesce': True,
            'max_instances': 10,
            'misfire_grace_time': 10
        },
        executors={
            'default': AsyncIOExecutor(),
            'threadpool': ThreadPoolExecutor(10),
            'processpool': ProcessPoolExecutor(2)
        }
    )
    app.state.scheduler = scheduler
    scheduler.add_job(job1, args=[app], trigger='interval', seconds=1)
    scheduler.add_job(job2, args=[app], trigger='interval', seconds=1)
    scheduler.start()

async def job1(app: FastAPI):
    # do time-consuming tasks

async def job1(app: FastAPI):
    # do something
```

**Comment by agronholm** [comment]
This is async 101. `AsyncIOExecutor` should only be used for async jobs. You already have alternative executors configured – why not use them for those blocking operations?

And next time, open a discussion instead of reporting a bug when you're asking for help on how to use the scheduler.

**Reply by agronholm** [reply]
You haven't shown what `job1` and `job2` do, and I assume you're running synchronous operations there that block the event loop. So perhaps use those other executors for those jobs, and make the target functions synchronous.
